sdk/metric/sdk.go

```diff
@@ -139,6 +143,15 @@ type (
 		labels        *label.Set
 		observed      export.Aggregator
 	}
+
+	// Implementations of MetricsProcessor can be provided as an config option to provide an opportunity
+	// to re-compose metrics labels based on the context when the metrics are recorded.
+	MetricsProcessor interface {
+		// OnMetricRecorded is execute everytime a metric is recorded by
+		// the sync instrument implementation of an Accumulator, it generally
+		// provides ability to correlate the context with the metrics
+		OnMetricRecorded(context.Context, *[]label.KeyValue)
```

Comment:
I'm guessing the added indirection here of a reference type for the second parameter means that argument will be mutated. This doesn't seem idiomatic of Go and it will likely add an additional allocation of the slice pointer and pose concurrency issues.

Is there any reason this could not accept a slice and return a slice?

Additionally, how are errors expected to be handled here?